Jenkinsfile

```diff
@@ -52,7 +52,8 @@ try {
       // Print the hostname to let us know on which node the docker image was executed for reproducibility.
       sh "hostname"
     }
-
+  
+  docker.withRegistry('https://registry.hub.docker.com', 'docker') {
```

Comment:
Do I understand it correctly that this is functionality offered by Jenkins? Are we logged in via the Jenkins UI and the necessary docker user data/token is taken from there?

Reply:
Jenkins offers a credentials store. This is where we already store, e.g., the Github credentials. I added a docker account to that store (a new, read-only user). `'docker'` is the ID of that credential. Jenkins internally does magic to retrieve the login information when needed. This is so that we don't have to store any credentials in the Jenkinsfile.